src/bft_membership.rs

```diff
@@ -0,0 +1,1326 @@
+use serde::{Deserialize, Serialize};
+use std::collections::{BTreeMap, BTreeSet};
+
+use crate::actor::{Actor, Sig, SigningActor};
+use crate::packet::{Packet, Payload};
+
+const SOFT_MAX_MEMBERS: usize = 7;
+pub type Generation = u64;
+
+#[derive(Debug, Default)]
+pub struct State {
+    pub id: SigningActor,
+    pub gen: Generation,
+    pub pending_gen: Generation,
+    trusted_members: BTreeSet<Actor>,
+    pub history: BTreeMap<Generation, Vote>, // for onboarding new procs, the vote proving quorum
+    pub votes: BTreeMap<Actor, Vote>,
+    pub faulty: bool,
+}
+
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
+pub enum Reconfig {
+    Join(Actor),
+    Leave(Actor),
+}
+
+impl std::fmt::Debug for Reconfig {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Reconfig::Join(a) => write!(f, "J{:?}", a),
+            Reconfig::Leave(a) => write!(f, "L{:?}", a),
+        }
+    }
+}
+
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
+pub enum Ballot {
+    Propose(Reconfig),
+    Merge(BTreeSet<Vote>),
+    Quorum(BTreeSet<Vote>),
```

Comment:
As discussed in HO, maybe we should do a global rename of Quorum to SuperMajority to add some clarity.

I believe we are working with these definitions:

Quorum: "we have enough members present to conduct a vote"
Majority: Greater than 50% of members
SuperMajority: Greater than 2/3 of members.

Well, according to Wikipedia, a SuperMajority has other common meanings such as 3/5 or 55% and can mean pretty much anything over 50%, but I think for our purposes we can just define it to mean 2/3.